### PR TITLE
feat(dynamodb): expose table's keys (hash and range)

### LINKED ISSUE #3215: DynamoDB: Allow accessing the table's key schema
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* **I'm submitting a ...**
  - [x] :rocket: feature request

* **What is the expected behavior (or behavior of feature suggested)?**

It would be nice if it was possible to read a DynamoDB table's key schema (hash attribute name and type, as well as range attribute name and type if there is one).

* **What is the motivation / use case for changing the behavior or adding this feature?**

When building abstractions on top of DynamoDB (think: API Gateway proxies, Lambda functions that perform scheduled maintenance, ...), it can be useful to know the key schema in order to be able to generate correct integration requests.

### PR DESCRIPTION
**Pull request by RomainMuller**
Adds attributes to read the attributes from the key schema:
- `tablePartitionKey`
- `tableSortKey`

Also made it possible to import a DynamoDB Table by introducing the
`ITable` interface as well as `Table.fromTableAttributes` method.

Fixes #3215

----

Please read the [contribution guidelines](https://github.com/awslabs/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-dynamodb/lib/table.ts [hunk 1]
```diff
@@ -181,10 +181,249 @@ export interface LocalSecondaryIndexProps extends SecondaryIndexProps {
...
+   * @attribute
+   */
+  readonly tableStreamArn: string;
+
+  /**
+   * The partition (hash) key of the Table.
+   *
+   * @attribute
+   */
+  readonly tablePartitionKey: Attribute;
```
Comment (eladb) [inline packages/@aws-cdk/aws-dynamodb/lib/table.ts]: Those should not be available here. They should only be available on the concrete construct and not for imported constructs.

Comment (skinny85) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts]: Yeah, this is an interesting discussion. I guess making it required means we can't ever have a `Table.fromTableArn` / `Table.fromTableName` method. But maybe there are other properties, like `tableStreamArn`, that already preclude it? If so, maybe we should just bite the bullet, and make them required here.

@RomainMuller your thoughts on the topic?

Comment (RomainMuller) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts]: Thing is it's very hard to actually *use* a DynamoDB table without knowing what it's key schema is... And if we want to have constructs that are "generic" like a "DynamoDBTableVIewer", having the key schema available makes things a lot nicer and easier. You don't have to duplicate the information about column names & types, ...

Comment (skinny85) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts]: OK, I'm fine with having these in `ITable` then. @eladb ?

Comment (eladb) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts]: No, I don't feel comfortable expanding that surface area, especially given we want to expose all the configuration props of all resources in their concrete class. `ITable` is essentially a wrapper to the `tableArn` (and any methods that can operate at that level). It's mostly needed for **cross-app** scenarios and quite rare.

Comment (RomainMuller) [reply packages/@aws-cdk/aws-dynamodb/lib/table.ts]: Alright then - so I'm discarding this PR.

**Pull request closed without merging**